llvm/include/llvm/Analysis/TargetLibraryInfo.h

```diff
@@ -249,6 +250,12 @@ class TargetLibraryInfoImpl {
   /// Returns the size of the size_t type in bits.
   unsigned getSizeTSize(const Module &M) const;
 
+  /// Returns an IntegerType corresponding to size_t.
+  IntegerType *getSizeTType(const Module &M) const;
+
+  /// Returns a constant materialized as a size_t type.
+  ConstantInt *getAsSizeT(uint64_t V, const Module &M) const;
```

Comment:
I don't think adding these to Impl is necessary, you can directly define these on TargetLibraryInfo (by calling Impl->getSizeTSize).

Reply:
Done